bindings/kepler.gl-jupyter/DEVELOPMENT.md

```diff
@@ -0,0 +1,205 @@
+The actual python code is in `./keplergl`.
+This provides a jupyter widget (a `ipywidgets.DOMWidget`).
+
+This python package contains and makes use of the javascript package `keplergl-jupyter`, which resides in [`./js`](./js).
+It provides a [JupyterLab extension](https://jupyterlab.readthedocs.io/en/stable/extension/extension_dev.html) (see `./js/labplugin.js`) and a [Jupyter Notebook extension](https://nbclassic.readthedocs.io/en/latest/extending/index.html) (see `./js/extension.js`) which share all the 'functional' code to display the kepler.gl-widget.
+
+When building the js code, this will generate javascript files in `./keplergl/static`.
+When installing the python package, the code in `./js` is also automatically built (unless the build results are already there, i.e. if you manually built it).
+
+Some example notebooks can be found in `./notebooks`.
+
+
+### Release
+
+See [`RELEASE.md`](RELEASE.md) for how this package is released.
+
+
+## Setup
+
+### Environment Setup
+You will need to install node, yarn and Jupyter Notebook.
+
+#### 1. Node and Yarn
+We recommended to use [`nvm`](https://github.com/nvm-sh/nvm) to install `node` and to use `yarn` via `corepack`:
+
+    cd js
+    nvm install  # this will install `node` in the version specified in `.nvmrc`
+    corepack enable
+    yarn  # this will install yarn in the version specified in `yarn.lock`.
+
+It may work to use other [node](https://nodejs.org/en/download/package-manager) `> 12` and [yarn](https://yarnpkg.com/getting-started/install) installations.
+
+#### Jupyter
+
+
+- Using conda
+```shell
+conda install jupyter
+conda install notebook 6.0.1
+```
+
+- Using pip
+  (When working with `pip` you may want to first create a virtual environment via `python -m venv venv && source ./venv/bin/activate` -- on Ubuntu you will need to install `python3-venv` for that)
+```shell
+pip install jupyter
+pip install notebook==6.0.1
+```
+
+#### 3. Install GeoPandas
+
+- Using conda
+```shell
+conda install geopandas
+```
+
+- Using pip
+
+```shell
+pip install geopandas
+```
+
+### Download and run keplergl in your local Jupyter Notebook
+
+#### Clone Repo
+```shell
+git clone https://github.com/keplergl/kepler.gl.git
+```
+
+### Setup JS
+#### 1. Install Js dependencies
+```sh
+cd bindings/kepler.gl-jupyter
+cd js
+yarn
+```
+
+Rerun `yarn` if any js dependency has changed (usually after pulling new changes from master).
+
+#### 2. Load mapbox token
+Add a [Mapbox access token](https://docs.mapbox.com/help/how-mapbox-works/access-tokens/) to the environment.
+
+```sh
+export MapboxAccessTokenJupyter=<your_mapbox_token>
+```
+
+#### 3. Build the js module
+
+For development of the js code, it is convenient to start a local server to watch for changes:
+
+```shell
+yarn start
+```
+
+If you edit a file in the `js` directory, the local server will pick up those changes.
+
+To build without a local server that watches for changes, use
+
+```shell
+yarn build
```

Comment:
this does not work for me -- should it?

Reply:
when I use `yarn start`, I end up with a labextension that errors out with 'Failed to load model class 'KeplerGlModal' from module 'keplergl-jupyter'`, is that because something is missing in the build?